CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/parameters/types/ElementTypeParameterValue.java

```diff
@@ -1,85 +1,113 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.parameters.types;
+
+import au.gov.asd.tac.constellation.graph.GraphElementType;
+import java.util.Objects;
+
+/**
+ * Element Type Parameter Value
+ *
+ * @author cygnus_x-1
+ */
+public class ElementTypeParameterValue extends ParameterValue {
+
+    private GraphElementType elementType;
+
+    public ElementTypeParameterValue() {
+        this.elementType = null;
+    }
+
+    public ElementTypeParameterValue(final GraphElementType elementType) {
+        this.elementType = elementType;
+    }
+
+    public GraphElementType getGraphElementType() {
+        return elementType;
+    }
+
+    @Override
+    public String validateString(final String s) {
+        try {
+            GraphElementType.getValue(s);
+        } catch (IllegalArgumentException ex) {
+            return String.format("%s is not a valid element type", s);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean setStringValue(final String s) {
+        final GraphElementType stringElementType = GraphElementType.getValue(s);
+        final boolean equal = Objects.equals(stringElementType, elementType);
+        if (!equal) {
+            elementType = stringElementType;
+        }
+        return equal;
+    }
+
+    @Override
+    public Object getObjectValue() {
+        return elementType;
+    }
+
+    @Override
+    public boolean setObjectValue(final Object o) {
+        if (o instanceof GraphElementType) {
+            final GraphElementType type = (GraphElementType) o;
+            final boolean equal = Objects.equals(type, elementType);
+            if (!equal) {
+                elementType = type;
+            }
+            return equal;
+        }
+        return false;
+    }
+
+    @Override
+    protected ParameterValue createCopy() {
+        return new ElementTypeParameterValue(elementType);
+    }
+
+    @Override
+    public String toString() {
+        return elementType == null ? "No Value" : elementType.getShortLabel();
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = 5;
+        hash = 41 * hash + Objects.hashCode(this.elementType);
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;

```

Comment:
Given this if statement and the above if statement have the same code block, you could probably merge the 2 conditions together with `||`. On a similar note, you can probably turn the if statement into an else if given that the 1st if statement just returns true

Reply:
This was auto generated code when I did an insert for the hash and equals functions so I didn't look too closely at it.

Reply:
I see what you mean, but after typing in those suggestions I actually think the code is slightly less readable then it is currently. So I am tempted to leave it as it is if that's okay?

Reply:
Having typed it out myself, I agree that it looks slightly less readable but I don't think it's THAT much less readable so I think it may still be worth doing it. Thoughts @arcturus2?

Note: If this is auto-generated code then if we decide to make changes, perhaps they need to be made further up the chain

Reply:
Hey @antares1470 my preference would be to keep the auto generated version and actually hide these blocks of code using code folding in NetBeans, see http://wiki.netbeans.org/SurroundWithCodeFolding. I've documented this idea in #821.